### PR TITLE
[msal-browser] resources and scopes documentation

### PR DESCRIPTION
**Pull request by derisen**
docs: adding a new doc for msal-browser

This PR adds a documentation on working with multiple resources and scopes in response to #1361 and #1653

### CODE REVIEW

File: lib/msal-browser/docs/resources-and-scopes.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          };
+
+          // will return an ID Token and an Access Token with scopes: openid, profile, User.Read
+          msalInstance.loginPopup(loginRequest);
+
+          // will fail and fallback to an interactive method prompting a consent screen
+          msalInstance.acquireTokenSilent(tokenRequest);
+     ```
+
+In the code snippet above, the user will prompted for consent once they authenticate and receive an **ID Token** and an **Access Token** with scope `User.Read`. Later, if they request an **Access Token** for `User.Read`, they will not be asked for consent again (in other words, they can acquire a token *silently*). On the other hand, the user did not consented to `Mail.Read` at the authentication stage. As such, they will be asked for consent when requesting an **Access Token** for that scope.
```
Comment (jo-arroyo) [inline lib/msal-browser/docs/resources-and-scopes.md:60]:
```suggestion
In the code snippet above, the user will  be prompted for consent once they authenticate and receive an **ID Token** and an **Access Token** with scope `User.Read`. Later, if they request an **Access Token** for `User.Read`, they will not be asked for consent again (in other words, they can acquire a token *silently*). On the other hand, the user did not consented to `Mail.Read` at the authentication stage. As such, they will be asked for consent when requesting an **Access Token** for that scope.
```

Comment (derisen) [reply lib/msal-browser/docs/resources-and-scopes.md]: fixed!

File: lib/msal-browser/docs/resources-and-scopes.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
+# Resources and Scopes
+
+> :warning: Before you start here, make sure you understand how to [acquire and use an access token](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/acquire-token.md).
+
+Azure Active Directory v2.0 & Microsoft Identity Platform employs a *scope-centric* model to access resources. Here, a *resource* refers to any application that can be of recipient of an **Access Token** (such as [MS Graph API](https://docs.microsoft.com/graph/overview) or your own web API), and a *scope* (*aka* "permission") refers to any aspect of a resource that an **Access Token** grants rights.
+
+**Access Token** requests in **MSAL.js** are meant to be *per-resource-per-scope(s)*. This means that an **Access Token** requested for resource **A** with scope `scp1`:
+
+- cannot be used for accessing resource **A** with scope `scp2`, and,
```
Comment (tnorling) [inline lib/msal-browser/docs/resources-and-scopes.md:9]: Add a note or expand the example to show that a single access_token can span multiple scopes across a shared resource

File: lib/msal-browser/docs/resources-and-scopes.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          };
+
+          // will return an ID Token and an Access Token with scopes: openid, profile, User.Read
+          msalInstance.loginPopup(loginRequest);
+
+          // will fail and fallback to an interactive method prompting a consent screen
+          msalInstance.acquireTokenSilent(tokenRequest);
+     ```
+
+In the code snippet above, the user will be prompted for consent once they authenticate and receive an **ID Token** and an **Access Token** with scope `User.Read`. Later, if they request an **Access Token** for `User.Read`, they will not be asked for consent again (in other words, they can acquire a token *silently*). On the other hand, the user did not consented to `Mail.Read` at the authentication stage. As such, they will be asked for consent when requesting an **Access Token** for that scope.
```
Comment (tnorling) [inline lib/msal-browser/docs/resources-and-scopes.md:60]: > On the other hand, the user did not consented to `Mail.Read` at the authentication stage. As such, they will be asked for consent when requesting an **Access Token** for that scope.

Note that after consenting they will receive a new access token containing both `Mail.Read` and the previously consented `User.Read`, which is what we mean by "incremental consent"

File: lib/msal-browser/docs/resources-and-scopes.md (hunk 2 above)
Comment (derisen) [reply lib/msal-browser/docs/resources-and-scopes.md:9]: added!

File: lib/msal-browser/docs/resources-and-scopes.md (hunk 3 above)
Comment (derisen) [reply lib/msal-browser/docs/resources-and-scopes.md]: added!

**Comment by derisen** [comment]
@tnorling thanks! Added a section on consent lifetime.

**Comment by tnorling** [comment]
Due to build failure and package-lock file we've copied the doc to #2002 and will pull into dev from there

**Pull request closed without merging**